### DISCUSSION TITLE
Heads up! I'm fixing missing MESS synths bug in pre 4

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1139] [Team Posts]
Missing fluidsynth, simpledrums, and vam. Each has undefined ref to MusEGlobal.config.
Caused by icon module changes.
Revealed a couple of old problems... The plot thickens... Stay tuned...

**Comment by terminator356** [comment]
It's late so I'll just post my log for convenience, and ask for a check of something for kybos:
In icons.cpp, it looks like MICONSVG can leave some icons undeleted
 if the user chooses theme icons. It looks to me like the examples given in
 MICON and MPIXMAP would do the job correctly if followed. Is it correct? Thanks.

  - HOT Bug fix: Missing 3 MESS synths, undefined symbol MusEGlobal::config (Tim)
    Cause: #include "gconfig.h" in icons.cpp brought in core globals which should
     not be in these synth plugin libraries (or in icons module). Removed the line
     and instead passed the desired global value to initIcons().
    That said, the plugins really should be stand alone and not use the icons module
     until the icons module becomes a true shared resource library. (Currently it
     is initialized by core, so not quite self sufficient yet.) So... I copied the
     required icons to the relevant synths, decoupling them from the icons module.
    By using icons directly from qrc instead of icons module, I was also able to
     remove the icons module from inclusion in widgets module (yay!), removing
     a further dependency of simpledrums synth which was missing too.
  * Module independence is a goal to strive for after 28.01.2018 MAJOR changes.

**Comment by kybos** [comment]
OK, I will check...
